etl/connection/grpc_client.go

```diff
@@ -90,3 +92,20 @@ func GetClientConfiguration(addr string) configuration.Replicator {
 		MaxTransportMsg: 100500,
 	}
 }
+
+func (c *GRPCClientConnection) NotifyShutdown(ctx context.Context, stopChannel chan<- struct{}, waitForStateChange time.Duration) {
+	go func(ctx context.Context, conn *grpc.ClientConn, stopChannel chan<- struct{}, waitForStateChange time.Duration) {
+		for {
+			if conn.GetState() == connectivity.TransientFailure {
+				ctx, cancel := context.WithTimeout(ctx, waitForStateChange)
+				defer cancel()
+				if !conn.WaitForStateChange(ctx, connectivity.TransientFailure) {
+					belogger.FromContext(ctx).Error("GRPC connection failed. Status is ", conn.GetState())
+					stopChannel <- struct{}{}
+					return
+				}
+			}
+			time.Sleep(time.Second * 10)
```

Comment:
will tests run for 10 seconds ?
may be this is a good place for  WaitForConnectionRecoveryTimeout
and you can use zero here for tests